src/main/kotlin/next/step/lotto/domain/LottoRank.kt

```diff
@@ -1,18 +1,25 @@
 package next.step.lotto.domain
 
-enum class LottoRank(val matchCount: Int, val winnings: Int, val matchBonus: Boolean = false) {
+enum class LottoRank(val matchCount: Int, val winnings: Int) {
     MISS(0, 0),
     FIFTH(3, 5_000),
     FOURTH(4, 50_000),
     THIRD(5, 1_500_000),
-    SECOND(5, 30_000_000, true),
+    SECOND(5, 30_000_000),
     FIRST(6, 2_000_000_000);
 
+    fun shouldMatchBonus(): Boolean = this == SECOND
+
     companion object {
 
-        fun from(matchCount: Int, matchBonus: Boolean): LottoRank =
-            values().filter { it.matchBonus == matchBonus }.find { it.matchCount == matchCount }
-                ?: (values().find { it.matchCount == matchCount } ?: MISS)
+        fun from(matchCount: Int, matchBonus: Boolean): LottoRank {
+            return if (isSecond(matchBonus, matchCount)) SECOND
+            else values().filterNot { it.shouldMatchBonus() }.find { it.matchCount == matchCount } ?: MISS
+        }
+
```

Comment:
matchCount가 5일 때만 SECOND or THIRD 를 반환하고, 나머지는 find로 반환해보는 방법도 있겠네요!
아래 방법은 참고 정도만 해주셔도 좋습니다 :)
```suggestion
        fun from(matchCount: Int, matchBonus: Boolean): LottoRank {
            return when {
                matchCount == 5 && matchBonus -> SECOND
                matchCount == 5 && !matchBonus -> THIRD
                else -> values().find { it.matchCount == matchCount } ?: MISS
            }
       }
```

Reply:
@malibinYun 제시해주신 방법에서 로직이 한 눈에 잘 들어오기도 하고, when 구문도 써보고 싶기도 해서 이렇게 바꿔보겠습니다 :)